rpcs3/Emu/Cell/Modules/cellSpurs.cpp

```diff
@@ -3129,17 +3130,83 @@ s32 _cellSpursWorkloadFlagReceiver(ppu_thread& ppu, vm::ptr<CellSpurs> spurs, u3
 		}
 
 		res = CELL_OK;
-		return;
 	});
 
 	return res;
 }
 
 /// Set/unset the recipient of the workload flag
-s32 _cellSpursWorkloadFlagReceiver2()
+s32 _cellSpursWorkloadFlagReceiver2(ppu_thread& ppu, vm::ptr<CellSpurs> spurs, u32 wid, u32 is_set, u32 print_debug_output)
 {
-	UNIMPLEMENTED_FUNC(cellSpurs);
-	return CELL_OK;
+	cellSpurs.warning("_cellSpursWorkloadFlagReceiver2(spurs=*0x%x, wid=%d, is_set=%d, print_debug_output=0%d)", spurs, wid, is_set, print_debug_output);
+
+	if (!spurs)
+	{
+		return CELL_SPURS_POLICY_MODULE_ERROR_NULL_POINTER;
+	}
+
+	if (!spurs.aligned())
+	{
+		return CELL_SPURS_POLICY_MODULE_ERROR_ALIGN;
+	}
+
+	if (wid >= spurs->max_workloads())
+	{
+		return CELL_SPURS_POLICY_MODULE_ERROR_INVAL;
+	}
+
+	if ((spurs->wklEnabled.load() & (0x80000000u >> wid)) == 0u)
+	{
+		return CELL_SPURS_POLICY_MODULE_ERROR_SRCH;
+	}
+
+	if (spurs->exception)
+	{
+		return CELL_SPURS_POLICY_MODULE_ERROR_STAT;
+	}
+
+	atomic_fence_acq_rel();
+
+	s32 res = CELL_OK;
+
+	vm::atomic_op<true>(spurs->wklFlagReceiver, [&](u8& FlagReceiver)
+	{
+		if (is_set)
+		{
+			if (FlagReceiver != 0xff)
+			{
+				res = CELL_SPURS_POLICY_MODULE_ERROR_BUSY;
+				return;
+			}
+		}
+		else
+		{
+			if (FlagReceiver != wid)
+			{
+				res = CELL_SPURS_POLICY_MODULE_ERROR_PERM;
+				return;
+			}
+		}
+
+		if (is_set)
+		{
+			if (FlagReceiver == 0xff)
+			{
+				FlagReceiver = static_cast<u8>(wid);
+			}
+		}
+		else
+		{
+			if (FlagReceiver == wid)
+			{
+				FlagReceiver = 0xff;
+			}
+		}
```

Comment:
this seems redundant.
can't you just set FlagReceiver in the scopes above ?

Reply:
I just seen it and they way it is is much more readable.